pyomo/repn/plugins/lp_writer.py

```diff
@@ -0,0 +1,600 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2022
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import logging
+from io import StringIO
+from operator import itemgetter, attrgetter
+
+from pyomo.common.config import (
+    ConfigBlock,
+    ConfigValue,
+    InEnum,
+    document_kwargs_from_configdict,
+)
+from pyomo.common.gc_manager import PauseGC
+from pyomo.common.timing import TicTocTimer
+
+from pyomo.core.base import (
+    Block,
+    Objective,
+    Constraint,
+    Var,
+    Param,
+    Expression,
+    SOSConstraint,
+    SortComponents,
+    Suffix,
+    SymbolMap,
+    minimize,
+)
+from pyomo.core.base.component import ActiveComponent
+from pyomo.core.base.label import LPFileLabeler, NumericLabeler
+from pyomo.opt import WriterFactory
+from pyomo.repn.linear import LinearRepnVisitor
+from pyomo.repn.quadratic import QuadraticRepnVisitor
+from pyomo.repn.util import (
+    FileDeterminism,
+    FileDeterminism_to_SortComponents,
+    categorize_valid_components,
+    initialize_var_map_from_column_order,
+    ordered_active_constraints,
+)
+
+### FIXME: Remove the following as soon as non-active components no
+### longer report active==True
+from pyomo.core.base import Set, RangeSet, ExternalFunction
+from pyomo.network import Port
```

Comment:
Just thinking out loud, but we really should fix this soon... We hack around it a lot